### PR TITLE
screenshot: Fix screenshot layer + trim compatibility.

### PR DESCRIPTION
**Pull request by plohrmann**
Screenshot layer was not handling the case that the most recently created
CommandPool object was also being destroyed, and thus was trying to use a
destroyed object. The layer now keeps a most-recently-created list of
CommandPools, and will utilize the one that was most recently created and
not yet destroyed. If there are no available CommandPools, then the
screenshot layer will not generate a file.

Fixes issue #182 

Change-Id: I8f248b44272f6b414f6ef578b962a2584c200e62

### CODE REVIEW

File: layersvt/screenshot.cpp [hunk 1]
```diff
@@ -678,9 +678,17 @@ static void writePPM(const char *filename, VkImage image1) {
 
     // Set up the command buffer.  We get a command buffer from a pool we saved
     // in a hooked function, which would be the application's pool.
+    if (deviceMap[device]->commandPools.empty())
+    {
```
Comment (ardaLunarG) [inline layersvt/screenshot.cpp]: Does not fit to clang here

Comment (plohrmann) [inline layersvt/screenshot.cpp]: Will fix.

File: layersvt/screenshot.cpp [hunk 2]
```diff
@@ -678,9 +678,16 @@ static void writePPM(const char *filename, VkImage image1) {
 
     // Set up the command buffer.  We get a command buffer from a pool we saved
     // in a hooked function, which would be the application's pool.
+    if (deviceMap[device]->commandPools.empty()) {
+        assert(!deviceMap[device]->commandPools.empty());
```
Comment (davidlunarg) [inline layersvt/screenshot.cpp]: Do we want this assert here? If there isn't a command pool to use, this will crash the process when compiled Debug, but will silently do nothing in Release mode.  Maybe the behavior should be the same for Release/Debug?

File: layersvt/screenshot.cpp [hunk 3]
```diff
@@ -1207,16 +1235,16 @@ VKAPI_ATTR VkResult VKAPI_CALL QueuePresentKHR(VkQueue queue, const VkPresentInf
...
-                for (auto it = imageMap.begin(); it != imageMap.end(); it++) {
-                    ImageMapStruct *imageMapElem = it->second;
+                for (auto imageIter = imageMap.begin(); imageIter != imageMap.end(); imageIter++) {
+                    ImageMapStruct *imageMapElem = imageIter->second;
                     delete imageMapElem;
                 }
-                for (auto it = physDeviceMap.begin(); it != physDeviceMap.end(); it++) {
-                    PhysDeviceMapStruct *physDeviceMapElem = it->second;
+                for (auto physDeviceIter = physDeviceMap.begin(); physDeviceIter != physDeviceMap.end(); physDeviceIter++) {
+                    PhysDeviceMapStruct *physDeviceMapElem = physDeviceIter->second;
```
Comment (davidlunarg) [inline layersvt/screenshot.cpp]: Why these changes renaming the iterator variable in this pull request? Seems unrelated to the gist of this pull request. But at this point, leave the changes in.

File: layersvt/screenshot.cpp (hunk 2 above)
Comment (plohrmann) [inline layersvt/screenshot.cpp]: I put the assert in because all the previous errors in this function which return, also have an assert. I agree they will have different behavior in release and debug, but was just following the existing behavior.

File: layersvt/screenshot.cpp (hunk 3 above)
Comment (plohrmann) [inline layersvt/screenshot.cpp]: Right it's unrelated to this change, but I saw a build warning related to the duplicate used variable name (there's already an iterator called 'it', so changed this one to ensure there wasn't any conflicts.

File: layersvt/screenshot.cpp (hunk 2 above)
Comment (davidlunarg) [inline layersvt/screenshot.cpp]: OK. Better to be consistent with existing code.
